### PR TITLE
Hypershift operator: Give a priority that is higher than any controlplane component

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/install/assets/hypershift_operator.go [hunk 1]
```diff
@@ -909,6 +913,18 @@ func (o HyperShiftEtcdPriorityClass) Build() *schedulingv1.PriorityClass {
 	}
 }
 
+func HypershiftOperatorPriorityClass() *schedulingv1.PriorityClass {
```
Comment (sjenning) [inline cmd/install/assets/hypershift_operator.go:916]: diverging from the
```
type HyperShiftEtcdPriorityClass struct{}
func (o HyperShiftEtcdPriorityClass) Build() *schedulingv1.PriorityClass {
...
```

pattern here.  Would like to keep them all the same.

Comment (alvaroaleman) [reply cmd/install/assets/hypershift_operator.go:916]: Id honestly rather change them all. That kind of pattern of defining an unused static object to then call a method on it is reserved for java code

**Comment by sjenning** [comment]
waiting for test run to complete to ensure that the priority is being created and set on the HO operator

File: cmd/install/assets/hypershift_operator.go (hunk 1 above)
Comment (ironcladlou) [reply cmd/install/assets/hypershift_operator.go:910]: > Id honestly rather change them all. That kind of pattern of defining an empty static object to then call a method on it is reserved for java code

shots fired 🤣

**Comment by enxebre** [comment]
may be mention this here https://hypershift-docs.netlify.app/how-to/distribute-hosted-cluster-workloads/#priority

File: cmd/install/assets/hypershift_operator.go [hunk 2]
```diff
@@ -909,6 +907,18 @@ func (o HyperShiftEtcdPriorityClass) Build() *schedulingv1.PriorityClass {
...
+	return &schedulingv1.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: HypershiftOperatorPriortyClass,
+		},
+		// Same as openshift-user-critical, which might not exist on plain kube clusters
+		Value:         1000000000,
```
Comment (sjenning) [inline cmd/install/assets/hypershift_operator.go:916]: do we want to be the same as `openshift-user-critical` or just use the next priority in our schema `100003000`

**Comment by sjenning** [comment]
for the record, the HO pod did get the priority set correctly :+1:

File: cmd/install/assets/hypershift_operator.go (hunk 2 above)
Comment (alvaroaleman) [reply cmd/install/assets/hypershift_operator.go]: Updated it to 100003000 and added this to the docs, ptal

**Pull request merged**